### PR TITLE
Add versions to draft artifacts

### PR DESCRIPTION
**Pull request by elsaperelli**
# Summary
Part of our feedback on the [Measure Repository Service specification](https://build.fhir.org/ig/HL7/cqf-measures/measure-repository-service.html#artifact-versioning) was that the cardinality of `version` on CQFMMeasure and CQFMLibrary (derived from CQFMShareableLibrary and CQFMShareableMeasure) is 1..1. This means that we want our draft artifacts to have versions on the authoring side as well as the artifacts on the published measure repository. This PR removes the version input part of the Release Modal and increments the artifact version when it is being converted to a draft rather than deleting it.

## New behavior
- Versions of existing artifacts that are being converted to a draft are now incremented rather than deleted.
- The user can no longer specify a version when releasing a draft artifact since that draft will already have one. 

## Code changes
- `app/package.json`/`package-lock.json` - installed `semver` for the inc and coerce functions
- `ReleaseModal.tsx` - removed the version user input
- `authoringFixtures.ts` - included a version of `0.0.1` on the Measure and Library skeletons. I decided to go with semantic versioning format here since it is a SHOULD in the spec, but we can change this to `0.0.001` if that makes more sense.
- `modifyResourceFields.ts` - increment the version of an existing artifact rather than deleting it. Added a TODO comment (feel free to remove) but I think it is something we should discuss and potentially make a task for?

# Testing guidance
- `npm run check:all` 

## Frontend Testing
- `npm run start:all`
- `localhost:3001`
- Create an artifact from scratch. Verify that the version in the JSON is `0.0.1`.
- Create an artifact from an existing artifact. Verify that the version in the JSON is incremented from the existing artifact. For example, if you have an artifact with version 0.0.003, then the draft artifact that you create from that should be 0.0.004.
- Release a draft artifact. Confirm that the version input is not there and that the released artifact has the version of the draft that it was. 

## Questions
- Should we update the typing at all? Right now, `FhirArtifact` is defined in `fhir.ts` as `fhir4.Measure | fhir4.Library`, whose versions have a cardinality of `0..1`. Should we somehow change this to `CQFMMeasure` and `CQFMLibrary` so the typing of `version` is accurate?
- Right now I am incrementing the version when converting an existing artifact to a draft artifact. Does that make sense? Should it instead be incremented when it is released to the Published Measure Repository?
- Should we be doing any handling to make sure that any artifacts being uploaded to the published measure repository have versions?

### CODE REVIEW

**Review by sarahmcdougall — CHANGES_REQUESTED** [review]
Left a few comments for suggested changes, but overall infrastructure looks good 👍 Great changes here!

File: app/src/util/modifyResourceFields.ts [hunk 1]
```diff
@@ -8,6 +9,64 @@ import { v4 as uuidv4 } from 'uuid';
...
-  delete artifact.version;
+  // we can only increment artifacts whose versions are either semantic, can be coerced
+  // to semantic, or are in x.x.xxx format. Every other kind of version will become 0.0.1
+  // TODO: right now we have FhirArtifact set to fhir4.Measure | fhir4.Library, those
+  // resources are defined with a version property whose cardinality is 0..1. The
+  // CQFMMeasure and CQFMLibrary resources have a version whose cardinality is 1..1.
+  // Do we want to change/modify this typing?
+  if (artifact.version) {
+    if (coerce(artifact.version) !== null) {
+      console.log('hi', artifact.version);
```
[the remark this answers is not in the document]
Comment (elsaperelli) [reply app/src/util/modifyResourceFields.ts:20]: LOL

File: app/src/util/modifyResourceFields.ts [hunk 2]
```diff
@@ -8,6 +9,64 @@ import { v4 as uuidv4 } from 'uuid';
...
+
+  // if the minor version reaches 10, reset it to 0 and increment the major version
+  if (versionParts[1] >= 10) {
+    versionParts[1] = 0;
+    versionParts[0]++;
+  }
+
+  let formattedPatch;
+  if (versionParts[2] < 10) {
+    formattedPatch = versionParts[2].toString().padStart(3, '0');
```
[the remark this answers is not in the document]
Comment (elsaperelli) [reply app/src/util/modifyResourceFields.ts:54]: Ikr! I was pretty pumped to find out about that

File: app/src/util/modifyResourceFields.ts [hunk 3]
```diff
@@ -8,6 +9,64 @@ import { v4 as uuidv4 } from 'uuid';
...
+  // we can only increment artifacts whose versions are either semantic, can be coerced
+  // to semantic, or are in x.x.xxx format. Every other kind of version will become 0.0.1
+  // TODO: right now we have FhirArtifact set to fhir4.Measure | fhir4.Library, those
+  // resources are defined with a version property whose cardinality is 0..1. The
+  // CQFMMeasure and CQFMLibrary resources have a version whose cardinality is 1..1.
+  // Do we want to change/modify this typing?
+  if (artifact.version) {
+    if (coerce(artifact.version) !== null) {
+      console.log('hi', artifact.version);
+      artifact.version = inc(artifact.version, 'patch') ?? '';
```
[the remark this answers is not in the document]
Comment (elsaperelli) [reply app/src/util/modifyResourceFields.ts:21]: I had to do some re-working of this anyways, so let me know if what I do now makes sense or if there should be any additional changes!

File: app/src/util/modifyResourceFields.ts [hunk 4]
```diff
@@ -8,6 +9,15 @@ import { v4 as uuidv4 } from 'uuid';
...
-  delete artifact.version;
+  // we can assume that there will be an artifact version since every artifact uploaded
+  // to the server gets a version if it does not already have one. We can also assume that
+  // it will be following semantic versioning since that handling is done in service
+  // TODO: right now we have FhirArtifact set to fhir4.Measure | fhir4.Library, those
+  // resources are defined with a version property whose cardinality is 0..1. The
+  // CQFMMeasure and CQFMLibrary resources have a version whose cardinality is 1..1.
+  // Do we want to change/modify this typing?
+  if (artifact.version) {
+    artifact.version = inc(artifact.version, 'patch') ?? '';
```
[the remark this answers is not in the document]
Comment (elsaperelli) [reply app/src/util/modifyResourceFields.ts]: We actually don't seem to have any documentation about the release functionality (or any authoring side functionality for that matter). Happy to do this but could also be a good EOS activity?

**Review by sarahmcdougall — APPROVED** [review]
Great changes here! Thanks for adding some comments surrounding the version incrementation process - I think those will be useful if we ever need to revisit this in the future. Functionality works great as well!

**Pull request merged**